modules/rubiconBidAdapter.js

```diff
@@ -93,32 +93,33 @@ export const spec = {
       return false;
     }
 
-    // Log warning if context is 'outstream', is not currently supported
-    if (utils.deepAccess(bid, `mediaTypes.${VIDEO}.context`) === 'outstream') {
-      utils.logWarn('Warning: outstream video for Rubicon Client Adapter is not supported yet');
-    }
-
-    // Log warning if mediaTypes contains both 'banner' and 'video'
-    if (spec.hasVideoMediaType(bid) && typeof utils.deepAccess(bid, `mediaTypes.${BANNER}`) !== 'undefined') {
-      utils.logWarn('Warning: instream video and banner requested for same ad unit, continuing with video instream request');
-    }
-
-    // Bid is invalid if legacy video is set but params video is missing size_id
-    if (bid.mediaType === 'video' && typeof utils.deepAccess(bid, 'params.video.size_id') === 'undefined') {
-      return false;
-    }
-
-    // Bid is invalid if mediaTypes video is invalid and a mediaTypes banner property is not defined
-    if (bid.mediaTypes && !spec.hasVideoMediaType(bid) && typeof bid.mediaTypes.banner === 'undefined') {
+    if (hasVideoMediaType(bid)) {
+      // Log warning if mediaTypes contains both 'banner' and 'video'
+      if (utils.deepAccess(bid, `mediaTypes.${VIDEO}.context`) === 'instream') {
+        if (typeof utils.deepAccess(bid, `mediaTypes.${BANNER}`) !== 'undefined') {
+          utils.logWarn('Warning: instream video and banner requested for same ad unit, continuing with video instream request');
+        }
+        // Bid is invalid if video is set but params video is missing size_id
+        if (typeof utils.deepAccess(bid, 'params.video.size_id') === 'undefined') {
+          utils.logError('Error: size id is missing for instream video request.');
+          return false;
+        }
+        return true;
+      } else if (utils.deepAccess(bid, `mediaTypes.${VIDEO}.context`) === 'outstream') {
+        if (utils.deepAccess(bid, 'params.video.size_id') !== 203) {
+          utils.logWarn('Warning: outstream video is sending invalid size id, converting size id to 203.');
+        }
+        return true;
+      }
+      // video context is neither instream nor outstream
+      utils.logError('Error: video context is neither instream nor outstream.');
       return false;
-    }
-
-    let parsedSizes = parseSizes(bid);
-    if (parsedSizes.length < 1) {
+    } else if (bid.mediaTypes && typeof utils.deepAccess(bid, `mediaTypes.${BANNER}`) === 'undefined') {
+      // Bid is invalid if mediaTypes video is invalid and a mediaTypes banner property is not defined
```

Comment:
`mediaTypes.banner` is not required, if there are no `mediaTypes` objects then we want to just default to banner. Not exit out of the bid request

Reply:
adUnit.sizes will eventually be deprecated, but we still are allowing it for now, and should continue to do so as long as Prebid Core does.